src/rtmp/dechunk.c

```diff
@@ -192,8 +192,29 @@ int rtmp_dechunker_receive(struct rtmp_dechunker *rd, struct mbuf *mb)
 		break;
 
 	case 3:
-		if (!chunk->mb)
-			return EPROTO;
+		if (!chunk->mb) {
+
+			chunk->mb = mbuf_alloc(chunk->hdr.length);
+			if (!chunk->mb)
+				return ENOMEM;
+
+			if (chunk->hdr.format == 0) {
```

Comment:
I suggest we simplify the timestamp logic a bit. 
```
if (!chunk->mb) {
	...
	if (chunk->hdr.format == 0)
		chunk->hdr.timestamp_delta = chunk->hdr.timestamp;

	chunk->hdr.timestamp += chunk->hdr.timestamp_delta;
}
```
Format must be one of: 0, 1 or 2 due to this logic:

```
switch (hdr.format) {
case 0:
case 1:
case 2:
	...
	chunk->hdr.format = hdr.format;
	break;
...
```